cocos/base/CCValue.cpp

```diff
@@ -491,6 +522,50 @@ int Value::asInt() const
     return 0;
 }
 
+
+unsigned int Value::asUnsignedInt() const
+{
+    CCASSERT(_type != Type::VECTOR && _type != Type::MAP && _type != Type::INT_KEY_MAP, "Only base type (bool, string, float, double, int) could be converted");
+    if (_type == Type::UNSIGNED)
+    {
+        return _field.unsignedVal;
+    }
+
+    if (_type == Type::INTEGER)
+    {
+        CCASSERT(_field.intVal >= 0, "Only values >= 0 can be converted to unsigned");
+        return static_cast<unsigned>(_field.intVal);
+    }
+
+    if (_type == Type::BYTE)
+    {
+        return static_cast<unsigned>(_field.byteVal);
+    }
+
+    if (_type == Type::STRING)
+    {
+        // NOTE: strtoul is required (need to augment on unsupported platforms)
+        return static_cast<unsigned int>(strtoul(_field.strVal->c_str(), NULL, 10));
+    }
+
+    if (_type == Type::FLOAT)
+    {
+        return static_cast<unsigned int>(_field.floatVal);
```

Comment:
Here uses static_cast<unsigned int>, but it uses static_cast<unsigned> in other places. I think they both the same thing, so it is better to use only one style.